controllers/provider-gcp/charts/internal/gcp-infra/templates/main.tf

```diff
@@ -31,6 +31,52 @@ resource "google_compute_subnetwork" "subnetwork-nodes" {
   region        = "{{ required "google.region is required" .Values.google.region }}"
 }
 
+{{ if .Values.create.cloudRouter -}}
+resource "google_compute_router" "router"{
+  name    = "{{ required "clusterName is required" .Values.clusterName }}-cloud-router"
+  region  = "{{ required "google.region is required" .Values.google.region }}"
+  network = "{{ required "vpc.name is required" .Values.vpc.name }}"
+}
+
+resource "google_compute_router_nat" "nat" {
+  name                               = "{{ required "clusterName is required" .Values.clusterName }}-cloud-nat"
+  router                             = "${google_compute_router.router.name}"
+  region                             = "{{ required "google.region is required" .Values.google.region }}"
+  nat_ip_allocate_option             = "AUTO_ONLY"
+  source_subnetwork_ip_ranges_to_nat = "LIST_OF_SUBNETWORKS"
+  subnetwork {
+    name                    =  "${google_compute_subnetwork.subnetwork-nodes.self_link}"
+    source_ip_ranges_to_nat = ["ALL_IP_RANGES"]
+  }
+
+  log_config {
+    enable = true
+    filter = "ERRORS_ONLY"
+  }
+}
+{{- end }}
+
+{{ if and (ne .Values.vpc.cloudRouter.name "none") (ne .Values.vpc.cloudRouter.name "")}}
```

Comment:
Don't check for name `none`. Rather use the standard pattern `{{ if .Values.vpc.cloudRouter.name ... }}`

Reply:
as we discussed, none would be better than empty.

Reply:
as we discussed, none would be better than empty.